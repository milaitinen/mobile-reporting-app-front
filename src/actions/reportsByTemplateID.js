// easier to refactor later and debug
export const STORE_REPORTS_BY_TEMPLATE_ID = 'STORE_REPORTS_BY_TEMPLATE_ID';

export const storeReportsByTemplateID = (reports) => ({
    type: STORE_REPORTS_BY_TEMPLATE_ID,
    reportsByTempID: reports
});
