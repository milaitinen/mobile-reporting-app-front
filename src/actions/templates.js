// easier to refactor later and debug
export const STORE_TEMPLATES = 'STORE_TEMPLATES';

export const storeTemplates = (templates) => ({
    type: STORE_TEMPLATES,
    templates: templates
});
