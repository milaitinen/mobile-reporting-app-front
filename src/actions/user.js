// easier to refactor later and debug
export const INSERT_EMAIL = 'INSERT_EMAIL';
export const INSERT_PASSWORD = 'INSERT_PASSWORD';
export const INSERT_SERVERURL = 'INSERT_SERVERURL';
export const ADD_TEMPLATES = 'ADD_TEMPLATES';

// return an object
export const insertEmail = (address) => ({
    type: INSERT_EMAIL,
    email: address
});

export const insertPassword = (password) => ({
    type: INSERT_PASSWORD,
    password: password
});

export const insertServerUrl = (url) => ({
    type: INSERT_SERVERURL,
    serverUrl: url
});

export const addTemplatesToData = (templates) => ({
    type: ADD_TEMPLATES,
    templates: templates
});