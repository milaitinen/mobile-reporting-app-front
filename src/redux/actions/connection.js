export const TOGGLE_CONNECTION = 'TOGGLE_CONNECTION';

export const toggleConnection = ({ connectionStatus }) => ({
    type: TOGGLE_CONNECTION,
    isConnected: connectionStatus,
});