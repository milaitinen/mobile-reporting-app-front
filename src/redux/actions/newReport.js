// easier to refactor later and debug
export const CREATE_REPORT  = 'CREATE_REPORT';
export const INSERT_TITLE   = 'INSERT_TITLE';

// return an object
export const createReport = ( templateID, isEditable) => ({
    type: CREATE_REPORT,
    templateID: templateID,
    isEditable: isEditable,
});

export const insertTitle = (title) => ({
    type: INSERT_TITLE,
    title: title
});
