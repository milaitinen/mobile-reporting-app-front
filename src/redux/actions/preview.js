// easier to refactor later and debug
export const PREVIEW = 'PREVIEW';
export const INSERT_TITLE = 'INSERT_TITLE';

// return an object
export const preview = ( templateID, isEditable) => ({
    type: PREVIEW,
    templateID: templateID,
    isEditable: isEditable,
});

export const insertTitle = (title) => ({
    type: INSERT_TITLE,
    title: title
});
