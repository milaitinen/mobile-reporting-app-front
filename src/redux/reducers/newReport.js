/* eslint-disable no-undef */
import {
    INSERT_TITLE,
    INSERT_DATE,
    INSERT_FIELD_ANSWER,
    EMPTY_FIELDS,
    SET_UNSAVED,
    OPEN_REPORT,
    CREATE_DRAFT
} from '../actions/newReport';

const initialState = {};

// A helper function to find the correct answer field inside the current state and update answer.
const insertAnswer = (state, action) => {
    // Check whether the field is option-type (dropdown, checkbox)
    if (action.isOption) {
        // Find answer field with the right id and update value
        if ((action.field.type === 'RADIOBUTTON') || (action.field.type === 'DROPDOWN')) { //clear selections if field type is radiobutton or dropdown
            state.option_answers.filter((answer) => {
                return (action.field.field_options
                    .map((option) => option.field_option_id)
                    .includes(answer.field_option_id));
            }).map((answer) => answer.selected = false);
        }

        state.option_answers.map(i => {
            if (i.field_option_id === action.value.field_option_id) i.selected = !i.selected;
        });
    } else {
        state.string_answers.map(i => {
            if (i.field_id === action.field.field_id) i.value = action.value;
        });
    }

    return {
        ...state
    };
};

const newReportReducer = (state = initialState, action) => {
    switch (action.type) {
        case INSERT_DATE: {
            return {
                ...state,
                date_created: action.date
            };
        }
        case INSERT_TITLE: {
            return {
                ...state,
                title: action.title
            };
        }
        case CREATE_DRAFT: {
            return action.draft;
        }
        case INSERT_FIELD_ANSWER: {
            return insertAnswer(state, action);
        }
        case OPEN_REPORT: {
            // Create deep copy, so that modifying the opened report doesn't affect the version saved in store.reports
            return JSON.parse(JSON.stringify(action.report));
        }
        case EMPTY_FIELDS: {
            return initialState;
        }
        case SET_UNSAVED: {
            return {
                ...state,
                isUnsaved: action.isUnsaved,
            };
        }
        default: {
            return state;
        }
    }
};

export default newReportReducer;
