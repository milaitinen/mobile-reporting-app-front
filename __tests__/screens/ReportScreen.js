import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'isomorphic-fetch';
//import { shallow } from 'enzyme';
//import { ActivityIndicator } from 'react-native';

import * as reportEditingTypes from '../../src/redux/actions/reportEditing';
import * as newReportTypes from '../../src/redux/actions/newReport';
import { ReportScreen } from '../../src/screens/ReportScreen';
//import { url } from '../../src/screens/urlsetting';

configure({ adapter: new Adapter() });

jest.mock('DatePickerIOS', () => 'DatePickerIOS');
Date.now = jest.fn(() => new Date(Date.UTC(2018, 4, 12)).valueOf());

const templates = {
    1: {
        template_id: 1,
        title: 'Template 1',
        fields: [
            {
                field_id: 1,
                template_id: 1,
                order_number: 1,
                title: 'Name',
                required: true,
                type: 'TEXTFIELD_SHORT',
                default_value: null,
                field_options: null
            }, {
                field_id: 2,
                template_id: 1,
                order_number: 2,
                title: 'Nice Checkbox',
                required: true,
                type: 'CHECKBOX',
                default_value: null,
                field_options: [
                    { field_option_id: 1, field_id: 2, value: 'Option 1', default_value: true },
                    { field_option_id: 2, field_id: 2, value: 'Option 2', default_value: false }
                ]
            }, {
                field_id: 3,
                template_id: 1,
                order_number: 3,
                title: 'Options',
                required: true,
                type: 'RADIOBUTTON',
                default_value: null,
                field_options: [
                    { field_option_id: 3, field_id: 3, value: 'Option 1', default_value: true },
                    { field_option_id: 4, field_id: 3, value: 'Option 2', default_value: false }
                ]
            }, {
                field_id: 4,
                template_id: 1,
                order_number: 4,
                title: 'Nice Dropdown',
                required: true,
                type: 'DROPDOWN',
                default_value: null,
                field_options: [
                    { field_option_id: 5, field_id: 4, value: 'Finland', default_value: true },
                    { field_option_id: 6, field_id: 4, value: 'Sweden', default_value: false }
                ]
            }, {
                field_id: 5,
                template_id: 1,
                order_number: 5,
                title: 'Instructions',
                required: true,
                type: 'INSTRUCTIONS',
                default_value: 'Very nice and long instruction',
                field_options: null
            }, {
                field_id: 6,
                template_id: 1,
                order_number: 6,
                title: 'Tell me about yourself',
                required: true,
                type: 'TEXTFIELD_LONG',
                default_value: null,
                field_options: null
            }, {
                field_id: 7,
                template_id: 1,
                order_number: 7,
                title: 'Calendar',
                required: true,
                default_value: '2018-04-12',
                type: 'CALENDAR'
            }, {
                field_id: 8,
                template_id: 1,
                order_number: 8,
                title: 'Time',
                required: true,
                default_value: '14:00',
                type: 'TIME'
            }, {
                field_id: 9,
                template_id: 1,
                order_number: 9,
                title: 'Google',
                required: true,
                default_value: 'http://www.google.com',
                type: 'LINK'
            }, {
                field_id: 10,
                template_id: 1,
                order_number: 10,
                title: 'Postal code',
                required: true,
                default_value: '3123',
                type: 'NUMBERFIELD'
            }
        ]
    }
};

const reports = {
    '1': [{
        report_id: 1,
        user_id: 1,
        template_id: 1,
        title: 'Testiraportti',
        date_created: '2018-03-16',
        date_accepted: null,
        string_answers: [
            { string_answer_id: 208, report_id: 1, field_id: 1, value: 'Test1' },
            { string_answer_id: 112, report_id: 1, field_id: 7, value: '2018-04-10' },
            { string_answer_id: 16, report_id: 1, field_id: 8, value: '10:33' },
            { string_answer_id: 200, report_id: 1, field_id: 9, value: 'http://www.google.com' },
            { string_answer_id: 200, report_id: 1, field_id: 10, value: '21789' }
        ],
        option_answers: [
            { option_answer_id: null, report_id: 1, field_option_id: 3, selected: true },
            { option_answer_id: null, report_id: 1, field_option_id: 4, selected: false }
        ]
    }]
};

const report = {
    report_id: 1,
    user_id: 1,
    template_id: 1,
    title: 'Testiraportti',
    date_created: '2018-03-16',
    date_accepted: null,
    string_answers: [
        { string_answer_id: 208, report_id: 1, field_id: 1, value: 'Test1' },
        { string_answer_id: 112, report_id: 1, field_id: 7, value: '2018-04-10' },
        { string_answer_id: 16, report_id: 1, field_id: 8, value: '10:33' },
        { string_answer_id: 200, report_id: 1, field_id: 9, value: 'http://www.google.com' },
        { string_answer_id: 200, report_id: 1, field_id: 10, value: '21789' }
    ],
    option_answers: [
        { option_answer_id: null, report_id: 1, field_option_id: 1, selected: true },
        { option_answer_id: null, report_id: 1, field_option_id: 2, selected: false },
        { option_answer_id: null, report_id: 1, field_option_id: 3, selected: true },
        { option_answer_id: null, report_id: 1, field_option_id: 4, selected: false },
        { option_answer_id: null, report_id: 1, field_option_id: 5, selected: false },
        { option_answer_id: null, report_id: 1, field_option_id: 6, selected: true }
    ]
};

describe('ReportScreen', () => {

    const dispatch = jest.fn();
    const goBack = jest.fn();
    const refresh = jest.fn();
    const removeDraft = jest.fn(() => new Promise(resolve => resolve()));

    const navigation = {
        goBack: goBack,
        dispatch: jest.fn(),
        state: { params: { templateID: 1, reportID: -1, isNewReport: false, title: 'Test', refresh: refresh, navigateDebounce: jest.fn() } }
    };

    const reportScreen = renderer.create(
        <ReportScreen
            navigation={navigation}
            isUnsaved={false}
            isSavingRequested={true}
            isNewReport={false}
            isConnected={true}
            reports={reports}
            report={report}
            templates={templates}
            dispatch={dispatch} />
    );

    jest.unmock('../../src/api');
    const saveToQueueWIthTemplateID = jest.fn();
    const api = require.requireActual('../../src/api');
    api.fetchEmptyTemplate = jest.fn(() => new Promise(resolve => resolve({ template_id: 2, report_id: null, user_id: 4 })));
    api.createNewReport = jest.fn(() => new Promise(resolve => resolve({ status: 200 })));
    api.removeDraft = removeDraft;
    api.saveToQueueWithTemplateID = saveToQueueWIthTemplateID;

    it('renders correctly', () => {
        const tree = renderer.create(
            <ReportScreen
                navigation={navigation}
                reports={reports}
                report={report}
                templates={templates}
                dispatch={dispatch}/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('saveAndLeave calls this.props.navigation.goBack()', () => {
        const inst = reportScreen.getInstance();
        inst.saveAndLeave();
        expect(goBack).toHaveBeenCalled();
    });

    it('send() calls this.props.navigation.state.params.refresh() when response status is 200', async () => {
        const inst = reportScreen.getInstance();
        await inst.send();
        expect(refresh).toHaveBeenCalled();
    });

    it('_createDraft should dispatch(createDraft(...))', async () => {
        const inst = reportScreen.getInstance();
        await inst._createDraft();
        expect(dispatch).toHaveBeenCalledWith({
            type: newReportTypes.CREATE_DRAFT,
            draft: { template_id: 2, report_id: null, user_id: 4 }
        });
    });

    it('_deleteDraft should empty fields by calling dispatch(emptyFields())', async () => {
        const inst = reportScreen.getInstance();
        await inst._deleteDraft();
        expect(dispatch).toHaveBeenCalledWith({
            type: newReportTypes.EMPTY_FIELDS
        });
    });

    it('insertAnswer() should dispatch setUnsaved(true) if this.props.isUnsaved=false', () => {
        const inst = reportScreen.getInstance();
        inst.insertAnswer('CHECKBOX', 'Helsinki', true);
        expect(dispatch).toHaveBeenCalledWith({
            type: reportEditingTypes.SET_UNSAVED,
            isUnsaved: true
        });
    });

    it('saveToPending() should call saveToQueueWIthTemplateID()', () => {
        const inst = reportScreen.getInstance();
        inst.saveToPending();
        expect(saveToQueueWIthTemplateID).toHaveBeenCalled();
    });

    it('dispatch(setSavingRequested(false)) should be called in componentWillUnmount if isSavingRequested=true and ' +
        'report_id is negative (draft)', () => {
        const inst = reportScreen.getInstance();
        inst.componentWillUnmount();

        expect(dispatch).toHaveBeenCalledWith({
            type: reportEditingTypes.SET_SAVING_REQUESTED,
            isSavingRequested: false
        });
    });

});