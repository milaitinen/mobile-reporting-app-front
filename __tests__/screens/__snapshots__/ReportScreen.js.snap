// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TemplateScreen renders correctly 1`] = `
<View
  colors={
    Array [
      undefined,
      undefined,
      undefined,
    ]
  }
  style={undefined}
>
  <View
    style={undefined}
  >
    <RCTScrollView
      keyboardShouldPersistTaps="handled"
      style={undefined}
    >
      <View>
        <View
          style={undefined}
        >
          <View
            style={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={undefined}
            >
              Otsikko
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={undefined}
            >
               *
            </Text>
          </View>
          <TextInput
            allowFontScaling={true}
            defaultValue="Testiraportti"
            editable={true}
            onChangeText={[Function]}
            placeholder={null}
            placeholderTextColor={null}
            style={undefined}
            underlineColorAndroid="transparent"
          />
        </View>
        <View
          pointerEvents={undefined}
        >
          <View
            style={undefined}
          >
            <View
              style={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={undefined}
              >
                Name
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={undefined}
              >
                 *
              </Text>
            </View>
            <TextInput
              allowFontScaling={true}
              onChangeText={[Function]}
              placeholder={null}
              placeholderTextColor={null}
              style={undefined}
              underlineColorAndroid="transparent"
              value="Test1"
            />
          </View>
          <View
            style={undefined}
          >
            <View
              style={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={undefined}
              >
                Options
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={undefined}
              >
                 *
              </Text>
            </View>
            <View
              style={
                Array [
                  undefined,
                  undefined,
                  false,
                ]
              }
            >
              <View
                style={undefined}
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "alignSelf": "center",
                        "flexDirection": "row",
                        "marginBottom": 5,
                      },
                      undefined,
                      false,
                    ]
                  }
                >
                  <View
                    style={undefined}
                  >
                    <View
                      accessibilityComponentType={undefined}
                      accessibilityLabel={undefined}
                      accessibilityTraits={undefined}
                      accessible={true}
                      collapsable={undefined}
                      hitSlop={undefined}
                      isTVSelectable={true}
                      nativeID={undefined}
                      onLayout={undefined}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "alignItems": "center",
                          "alignSelf": "center",
                          "borderColor": "#2196f3",
                          "borderRadius": 12,
                          "borderWidth": 1.5,
                          "height": 24,
                          "justifyContent": "center",
                          "opacity": 1,
                          "width": 24,
                        }
                      }
                      testID={undefined}
                      tvParallaxProperties={undefined}
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "borderRadius": 10,
                            },
                            Object {
                              "backgroundColor": "#2196f3",
                              "height": 20,
                              "width": 20,
                            },
                            Object {
                              "borderRadius": 8,
                              "height": 16,
                              "width": 16,
                            },
                            Object {
                              "backgroundColor": "#359ef3",
                            },
                          ]
                        }
                      />
                    </View>
                  </View>
                  <View
                    accessibilityComponentType={undefined}
                    accessibilityLabel={undefined}
                    accessibilityTraits={undefined}
                    accessible={true}
                    hitSlop={undefined}
                    nativeID={undefined}
                    onLayout={undefined}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Array [
                        undefined,
                        Object {
                          "alignItems": "center",
                          "alignSelf": "center",
                          "flexDirection": "row",
                        },
                      ]
                    }
                    testID={undefined}
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "lineHeight": 20,
                            "paddingLeft": 10,
                          },
                          false,
                          Object {
                            "color": undefined,
                          },
                          undefined,
                        ]
                      }
                    >
                      Option 1
                    </Text>
                  </View>
                </View>
              </View>
              <View
                style={undefined}
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "alignSelf": "center",
                        "flexDirection": "row",
                        "marginBottom": 5,
                      },
                      undefined,
                      false,
                    ]
                  }
                >
                  <View
                    style={undefined}
                  >
                    <View
                      accessibilityComponentType={undefined}
                      accessibilityLabel={undefined}
                      accessibilityTraits={undefined}
                      accessible={true}
                      collapsable={undefined}
                      hitSlop={undefined}
                      isTVSelectable={true}
                      nativeID={undefined}
                      onLayout={undefined}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "alignItems": "center",
                          "alignSelf": "center",
                          "borderColor": "#2196f3",
                          "borderRadius": 12,
                          "borderWidth": 1.5,
                          "height": 24,
                          "justifyContent": "center",
                          "opacity": 1,
                          "width": 24,
                        }
                      }
                      testID={undefined}
                      tvParallaxProperties={undefined}
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "borderRadius": 10,
                            },
                            false,
                            false,
                            false,
                          ]
                        }
                      />
                    </View>
                  </View>
                  <View
                    accessibilityComponentType={undefined}
                    accessibilityLabel={undefined}
                    accessibilityTraits={undefined}
                    accessible={true}
                    hitSlop={undefined}
                    nativeID={undefined}
                    onLayout={undefined}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Array [
                        undefined,
                        Object {
                          "alignItems": "center",
                          "alignSelf": "center",
                          "flexDirection": "row",
                        },
                      ]
                    }
                    testID={undefined}
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "lineHeight": 20,
                            "paddingLeft": 10,
                          },
                          false,
                          Object {
                            "color": undefined,
                          },
                          undefined,
                        ]
                      }
                    >
                      Option 2
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View>
          <View
            style={undefined}
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  undefined,
                ]
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <View
                style={undefined}
              >
                <View
                  onPress={undefined}
                  style={
                    Array [
                      false,
                      false,
                      false,
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      },
                      undefined,
                    ]
                  }
                  underlayColor="white"
                >
                  <Text
                    accessible={true}
                    allowFontScaling={false}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "black",
                          "fontSize": 18,
                        },
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                          },
                          undefined,
                        ],
                        Object {
                          "fontFamily": "Material Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                      ]
                    }
                  >
                    
                  </Text>
                </View>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={undefined}
                >
                  createNew.save
                </Text>
              </View>
            </View>
          </View>
          <View
            style={undefined}
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  undefined,
                ]
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <View
                style={undefined}
              >
                <View
                  onPress={undefined}
                  style={
                    Array [
                      false,
                      false,
                      false,
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      },
                      undefined,
                    ]
                  }
                  underlayColor="white"
                >
                  <Text
                    accessible={true}
                    allowFontScaling={false}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "black",
                          "fontSize": 18,
                        },
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                          },
                          undefined,
                        ],
                        Object {
                          "fontFamily": "Material Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                      ]
                    }
                  >
                    
                  </Text>
                </View>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={undefined}
                >
                  createNew.send
                </Text>
              </View>
            </View>
          </View>
          <View
            style={undefined}
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  undefined,
                ]
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <View
                style={undefined}
              >
                <View
                  onPress={undefined}
                  style={
                    Array [
                      false,
                      false,
                      false,
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      },
                      undefined,
                    ]
                  }
                  underlayColor="white"
                >
                  <Text
                    accessible={true}
                    allowFontScaling={false}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "black",
                          "fontSize": 18,
                        },
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                          },
                          undefined,
                        ],
                        Object {
                          "fontFamily": "Material Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                      ]
                    }
                  >
                    
                  </Text>
                </View>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={undefined}
                >
                  Delete
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
